app/views/posts/index.html.erb

```diff
@@ -0,0 +1,7 @@
+<% @posts.each do |p| %>
+<h3><%= p.title%></h3>
+<% if current_user %>
```

Comment:
- [ ] Rails has built-in mechanism to deal with view-related logic which is View Helpers (app/helpers/*_helper.rb). So the common convention is to move any logic in the views to the helpers. Please take a look at [this](https://mixandgo.com/learn/the-beginners-guide-to-rails-helpers) for more information.